src/Sulu/Bundle/ContentBundle/Resources/public/js/components/internal-links/main.js

```diff
@@ -232,6 +231,38 @@ define([], function() {
             return [this.options.url, delimiter, this.options.idsParameter, '=', (data || []).join(',')].join('');
         },
 
+        overlayOkCallback: function () {
+            this.sandbox.emit('husky.column-navigation.' + this.options.instanceName + '.get-marked', function (markedCollections) {
+                var toAdd = [],
+                    toRemove = [];
+
+                markedCollections.forEach(function(markedCollection) {
+                    if ($.inArray(markedCollection, this.getData()) < 0) {
+                        toAdd.push(markedCollection);
+                    }
+                }.bind(this));
+
+                var markedCollectionIds = [];
+                markedCollections.forEach(function (markedCollection) {
+                    markedCollectionIds.push(markedCollection.id);
+                }.bind(this));
+
+                this.getData().forEach(function(collection) {
+                    if ($.inArray(collection, markedCollectionIds) < 0) {
```

Comment:
Why not inverting this check? If we iterate over the `markedCollections` we could write something like `$.inArray(markedCollection.id, data)`. Would also remove the need for the previous loop?